packages/mattermost-redux/src/selectors/entities/preferences.ts

```diff
@@ -195,12 +195,14 @@ export function isCollapsedThreadsAllowed(state: GlobalState): boolean {
     );
 }
 
-export function isCollapsedThreadsEnabled(state: GlobalState): boolean {
-    const isAllowed = isCollapsedThreadsAllowed(state);
-    const userPreference = getCollapsedThreadsPreference(state);
-
-    return isAllowed && (userPreference === Preferences.COLLAPSED_REPLY_THREADS_ON || getConfig(state).CollapsedThreads as string === 'always_on');
-}
+export const isCollapsedThreadsEnabled = createSelector(
```

Comment:
Using `createSelector` here shouldn't be necessary. Since the results of the input selectors are just primitive values and a reference to the config, it probably takes about the same time to see if their results have been to check if their results have changes as it does to just return the result of that calculation. `createSelector` is only really needed if either the computation of the final result function is expensive or if it's returning an object/array

Reply:
yes makes sense since it would not be much performance gain. Reverted.